Python/Product/PythonTools/PythonTools/PythonClassifier.cs

```diff
@@ -19,58 +19,46 @@
 using System.Diagnostics;
 using System.Diagnostics.Contracts;
 using System.Text;
+using System.Threading.Tasks;
 using Microsoft.PythonTools.Editor;
 using Microsoft.PythonTools.Intellisense;
 using Microsoft.PythonTools.Interpreter;
 using Microsoft.PythonTools.Parsing;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
 
 namespace Microsoft.PythonTools {
     /// <summary>
     /// Provides classification based upon the DLR TokenCategory enum.
     /// </summary>
-    internal sealed class PythonClassifier : IClassifier {
+    internal sealed class PythonClassifier : IClassifier, IPythonTextBufferInfoEventSink {
         private readonly TokenCache _tokenCache;
         private readonly PythonClassifierProvider _provider;
-        private readonly PythonTextBufferInfo _buffer;
-        private PythonLanguageVersion _version;
 
         [ThreadStatic]
         private static Dictionary<PythonLanguageVersion, Tokenizer> _tokenizers;    // tokenizer for each version, shared between all buffers
 
-        internal PythonClassifier(PythonClassifierProvider provider, PythonTextBufferInfo buffer) {
-            buffer.OnChanged += BufferChanged;
-            buffer.OnContentTypeChanged += BufferContentTypeChanged;
-
+        internal PythonClassifier(PythonClassifierProvider provider) {
             _tokenCache = new TokenCache();
             _provider = provider;
-            _buffer = buffer;
-
-            _buffer.OnNewAnalysisEntry += NewAnalysisEntry;
-
-            _version = _buffer.LanguageVersion;
         }
 
-        private void NewAnalysisEntry(object sender, EventArgs e) {
-            var analyzer = _buffer.AnalysisEntry?.Analyzer;
+        private Task OnNewAnalysisEntryAsync(PythonTextBufferInfo sender, AnalysisEntry entry) {
```

Comment:
>OnNewAnalysisEntryAsync [](start = 21, length = 23)

Doesn't need to be async

Reply:
Also doesn't hurt, and it saves changes later if we make it async (it's been refactored out of an async method).